### ISSUE TITLE
Fetching remote-entry files dynamically and support of experiments.outputModule

### ISSUE THREAD

**Issue by harirsskrishna** [webpack/webpack#18119]
Hello Webpack contributors,

I've encountered an issue with dynamically loading remote-entry files (fetching  remote-entry files as needed) using Module Federation. I have a custom module loader that uses `__webpack_require__.l()` to fetch remote-entry files dynamically, as described in the [Module Federation Plugin documentation](https://webpack.js.org/plugins/module-federation-plugin).

To ensure `__webpack_require__.l()` is defined, I'm employing a workaround that involves dynamically importing a fake module (`await import('./FakeModule.js')`). While this successfully generates the `__webpack_require__.l()` definition, it has an unintended side effect: the `webpack/runtime/publicPath` (`__webpack_require__.p`) is set based on the baseURL of my module loader, rather than being recalculated for each use.

This has led to incorrect URLs for chunk loading, as `__webpack_require__.f.j()` calculates the chunk URL using the following pattern:

```js
var url = __webpack_require__.p + __webpack_require__.u(chunkId);
```
I've found that setting `output.library.type: 'module'` and enabling `experiments.outputModule` allows `__webpack_require__.p` to be set correctly using `import.meta.url`. However, this is not an ideal solution as it relies on experimental features.

I have two questions for the community:

1. I would like to avoid building my own script tag creation and eventing code, and utilize as much in model webpack code as possible.  

2. Are there any plans to support `experiments.outputModule` in future Webpack versions, or to stabilize this in the feature?

Any insights or suggestions on this matter would be greatly appreciated.

Thank you for your time and assistance.

Best regards,
Hari

**Comment by harirsskrishna** [comment]
@ScriptedAlchemy do you have any inputs?

**Comment by alexander-akait** [comment]
@harirsskrishna Please format your issue using the issue template, otherwise I will move it in `Dicussions`, thank you

**Comment by harirsskrishna** [comment]
@alexander-akait , I believe this better fits in the `Discussions` section. Please feel free to move it to `Discussions`.

**Comment by ScriptedAlchemy** [comment]
Look at module-federation/enhanced. I forked it out of webpack and added a hook lifecycle for runtime plugins.

**Comment by alexander-akait** [comment]
@ScriptedAlchemy Do we need something improve on webpack side?

**Comment by ScriptedAlchemy** [comment]
Im not sure if webpack will accept the enhancements made in here, it is also easier to maintain parity with webpack and rspack when both depend on a central runtime vs it coupled deeply to bundler. 

What i did is extract the runtime code from webpack and author a library, then i added the hooks on require as needed, like f.consumes etc to bind it back to the bundler runtime. 

If users want to adjust internals etc. You can write a runtime plugin that gives you access to the lifecycles.
for example: 
https://github.com/module-federation/module-federation-examples/blob/master/runtime-plugins/remote-control/app1/pick-remote.ts

Might be useful.

**Comment by alexander-akait** [comment]
@ScriptedAlchemy sounds intresting, it can be useful, we have hooks, so I think we can solve it by them

**Comment by ScriptedAlchemy** [comment]
indeed, im not sure what your opinion is on depending on module-federation/runtime - but we would always introcude this back into the webpack plugin, otherwise one can use the enhanced plugin
